src/entry.sh

```diff
@@ -572,99 +756,202 @@ function StartFHEM {
   export LC_PAPER="${LC_PAPER:-de_DE.UTF-8}" # Paper size in European standard
   export LC_TELEPHONE="${LC_TELEPHONE:-de_DE.UTF-8}" # Representation of telephone numbers in German format
   export LC_TIME="${LC_TIME:-en_DK.UTF-8}" # Date and time formats in European standard
-  export TZ="${TZ:-Europe/Berlin}"
-  [ "${LC_CTYPE}" != '' ] && export LC_CTYPE
-  [ "${LC_COLLATE}" != '' ] && export LC_COLLATE
-  [ "${LC_NAME}" != '' ] && export LC_NAME
-  [ "${LC_ADDRESS}" != '' ] && export LC_ADDRESS
-  [ "${LC_ALL}" != '' ] && export LC_ALL
+  [ -n "${LC_CTYPE+x}" ] && export LC_CTYPE
+  [ -n "${LC_COLLATE+x}" ] && export LC_COLLATE
+  [ -n "${LC_NAME+x}" ] && export LC_NAME
+  [ -n "${LC_ADDRESS+x}" ] && export LC_ADDRESS
+  [ -n "${LC_ALL+x}" ] && export LC_ALL
 
   # Export some variables someone might want to use
-  while IFS='=' read -r -d '' n v; do
-      [[ $n = 'NODE'* ]] && export "$n"
-      [[ $n = 'PERL'* ]] && export "$n"
-      [[ $n = 'PYTHON'* ]] && export "$n"
-  done < <(env -0)
+  for theVar in $(env | awk -F= '/^NODE|^PERL|^PYTHON/{print $1}'); do export $theVar ; done
+}
+
+
+#====================================================================================================================-
+#--- FHEM process functions ------------------------------------------------------------------------------------------
 
+# Read the PID of FHEM from it's PID file and write it to stdout
+# ATTENTION: The function returns only the number from the file. 
+#            No check is done if there in fact is a process with that number.
+#
+# Usage: getFhemPidNum
+# Global vars: PIDFILE
+# Returns:     0:     PID successfully read from file and sent to stdout
+#              else:  No PID available
+#
+function getFhemPidNum() {
+  [ -s "$PIDFILE" ] || return 1;
+  head -1 "$PIDFILE"
+  return 0
+}
+
+
+# Checks if there is a running process with the FHEM PID number.
+#
+# Usage: checkFhemProcessExists
+# Returns:     0:     Process exists
+#              else:  No process or any other error
+#
+function checkFhemProcessExists() {
+  local -i pidNum  # DO NOT make the following assignment in the same line! The return value gets lost!
+  pidNum=$(getFhemPidNum) || return
+  kill -0 $pidNum 2>&1 >/dev/null;
+}
+
+
+# Starts FHEM.
+# Main steps in here:
+#  - Run pre-start scripts
+#  - Start printing the logfile to the console
+#  - Create or patch Telnet Device for health check
+#  - Launch FHEM process
+#  - Run post-start scripts
+#
+# Usage: startFhemProcess
+# Global vars: lots of (tbd)
+#
+function startFhemProcess() {
+  runScript "/pre-start.sh"
+  runScript "/docker/pre-start.sh"
+
+  local realLogFile="$( date +"${LOGFILE}")"
+  touch "$realLogFile"                     # Make sure logfile exists, so tail stats with delay
+  chown ${FHEM_UID}:${FHEM_GID} "$realLogFile"
+  tailFileToConsoleStart "$realLogFile"    # Start writing the logfile to the console
+
+  # setTelnet_DEFINITION
```

Comment:
@HaraldR42 

I disabled the creation of a telnet device for docker, because i think with this improved healh check this is not needed anymore